src/coreComponents/physicsSolvers/fluidFlow/IsothermalCompositionalMultiphaseFVMKernels.hpp

```diff
@@ -717,37 +721,111 @@ class FaceBasedAssemblyKernel : public FaceBasedAssemblyKernelBase
           localIndex const esr_up = sesri[k_up];
           localIndex const ei_up  = sei[k_up];
 
-          real64 const mobility = m_phaseMob[er_up][esr_up][ei_up][ip];
-
-          // skip the phase flux if phase not present or immobile upstream
-          if( LvArray::math::abs( mobility ) < 1e-20 ) // TODO better constant
+          // C1PPU
+          if( m_useC1PPU > 0 && m_epsC1PPU > 0 )
           {
-            continue;
-          }
+            GEOS_ASSERT( numFluxSupportPoints == 2 );
```

Comment:
what would be the proper way to check?

Reply:
I think the most portable would be ``GEOSX_THROW_IF()`` but to confirm with @francoishamon that we check for that as ``numFluxSupportPoints`` is static constexpr so it would be known at compile time. Then throwing at runtime based on this value is suboptimal.

Reply:
In this part of the code, we have clearly decided that we will not use MPFA any time soon, and we have hard-coded a lot of specific values that are specific to TPFA, like `numFluxSupportPoints` [here](https://github.com/GEOS-DEV/GEOS/blob/672c180ed91324f063206bd12e6c1d85d2954e77/src/coreComponents/physicsSolvers/fluidFlow/IsothermalCompositionalMultiphaseFVMKernels.hpp#L419). So I would say that it is safe to remove this check entirely. If we want to add this check, we could add in the `createAndLaunch` or in the constructor of the kernel class.

Reply:
thank you
i removed the check